src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/SPARQL2GraphQLTranslatorImpl.java

```diff
@@ -0,0 +1,196 @@
+        final Map<Node, Set<TP>> subgraphPatterns = new HashMap<>();
+
+        // Partition BGP into SGPs and give unique integer id to TPs
+        final Set<? extends TriplePattern> bgpSet = bgp.getTriplePatterns();
+        int idCounter = 0;
+        for(final TriplePattern t : bgpSet){
+            final TP wrappedTriplePattern = new TP(idCounter,t);
+            final Node subject = t.asJenaTriple().getSubject();
+            if(subgraphPatterns.containsKey(subject)){
+                subgraphPatterns.get(subject).add(wrappedTriplePattern);
+            }
+            else{
+                final Set<TP> sgp = new HashSet<>();
+                sgp.add(wrappedTriplePattern);
+                subgraphPatterns.put(subject, sgp);
+            }
+            ++idCounter;
+        }
+
+        return subgraphPatterns;
+    }
+
+    /**
+     * Creates a connector map using @param subgraphPatterns
+     */
+    protected Map<Integer,Node> createConnectors(final Map<Node, Set<TP>> subgraphPatterns){
+        final Map<Integer,Node> connectors = new HashMap<>();
+        final Map<Node,SGPNode> sgpNodes = new HashMap<>();
+        for(final Node subject : subgraphPatterns.keySet()){
+            final Set<TP> sgp = subgraphPatterns.get(subject);
+            for(final TP tp : sgp){
+                final Node object = tp.getTriplePattern().asJenaTriple().getObject();
+                if(subgraphPatterns.containsKey(object) && ! object.equals(subject)){
+                    if(!sgpNodes.containsKey(subject)){
+                        sgpNodes.put(subject,new SGPNode());
+                    }
+                    if(!sgpNodes.containsKey(object)){
+                        sgpNodes.put(object,new SGPNode());
+                    }
+                    final SGPNode subjectSgpNode = sgpNodes.get(subject);
+                    final SGPNode objectSgpNode = sgpNodes.get(object);
+                    subjectSgpNode.addAdjacentNode(tp.getId(),objectSgpNode);
+                    connectors.put(tp.getId(), object);
+                }
+            }
+        }
+        // Remove all potential cyclic connector bindings
+        final Set<Integer> connectorsToBeRemoved = GraphCycleDetector.determineCyclicConnectors(sgpNodes);
+        connectors.keySet().removeAll(connectorsToBeRemoved);
+        return connectors;
+    } 
+
+    /**
+     * Generates query data for @param fieldPaths, @param queryArgs
+     */
+    protected void generateQueryData(final GraphQL2RDFConfiguration config, final GraphQLEndpoint endpoint,
+            final Map<Node,Set<TP>> subgraphPatterns, final Map<Integer,Node> connectors, final Set<String> fieldPaths,
+            final Set<GraphQLArgument> queryArgs, final Map<Node,String> withoutConnector){
+
+        // Counters for creating unique variable and entrypoint names
+        int entrypointCounter = 0;
+        final MutableInt variableCounter = new MutableInt(0);
```

Comment:
The reason this is not a regular integer is because of the method `addEntrypoint` in the `GraphQLFieldPathBuilder` where the entrypoint and arguments are added to fieldpaths. Its primary purpose is to make sure that the variables used in the arguments doesn't have the same name which means consecutive calls to `addEntrypoint` should still keep creating unique variable names. If it were a normal integer, the incrementing that is done in the loop of `addEntrypoint` after a new unique variable name is created wouldn't change the value of the integer outside of the method. Hope that explains it.